pkg/clients/openshift/client.go

```diff
@@ -99,14 +99,17 @@ func (c *Client) GetJobLogs(ctx context.Context, name, namespace string) (string
 // If the watched job fails, is disconnected, the watch produces an error, the
 // watch channel closes, or the context is cancelled at timeout, it will return
 // an error containing event in question.
-func (c *Client) WatchJob(ctx context.Context, namespace string, job batchv1.Job) error {
+func (c *Client) WatchJob(ctx context.Context, namespace string, name string) error {
+	job := new(batchv1.Job)
+	if err := c.Get(ctx, name, namespace, job); err != nil {
+		return fmt.Errorf("failed to find job %s/%s: %w", namespace, name, err)
+	}
 	clientSet, err := kubernetes.NewForConfig(c.GetConfig())
```

Comment:
What should we do if the job has more than 1 pod running in it? This makes the function very specific to our job we are testing (which does have a single pod running)

```
❯ oc explain job.status.succeeded
GROUP:      batch
KIND:       Job
VERSION:    v1

FIELD: succeeded <integer>

DESCRIPTION:
    The number of pods which reached phase Succeeded.
```

Reply:
I'm fine to add a short circuit, I have added one a bit more specific and notice how we don't have to add the `else { ... }` block because we return early (even in your above example, the else is not needed)